server.go

```diff
@@ -445,13 +448,12 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq, inbound
 
 	// Now that we've established a connection, create a peer, and
 	// it to the set of currently active peers.
-	p, err := newPeer(conn, connReq, s, peerAddr, inbound)
+	p, err := newPeer(conn, connReq, s, peerAddr, inbound, s.chanDB)
 	if err != nil {
 		srvrLog.Errorf("unable to create peer %v", err)
-		if p.connReq != nil {
-			s.connMgr.Remove(p.connReq.ID())
+		if connReq != nil {
+			s.connMgr.Remove(connReq.ID())
 		}
-		p.Disconnect()
```

Comment:
Ahh, nice find! In the future, I'd prefer for fixes like this to be either included in the PR in a distinct commit, or entirely within it's own PR. Otherwise, it's easy to miss amidst all the other changes within the PR.